From: Namhyung Kim <namhyung@gmail.com>
Subject: cfq-iosched: free cic_index if cfqd allocation fails
Git-commit: 1547010e6e15a3f44f49381246421a1e19de526e
Patch-mainline: yes
References: fate#312039

When struct cfq_data allocation fails, cic_index need to be freed.

Signed-off-by: Namhyung Kim <namhyung@gmail.com>
Signed-off-by: Jens Axboe <jaxboe@fusionio.com>
Acked-by: Jan Kara <jack@suse.cz>
---
 block/cfq-iosched.c |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/block/cfq-iosched.c b/block/cfq-iosched.c
index 891b176..387e9f4 100644
--- a/block/cfq-iosched.c
+++ b/block/cfq-iosched.c
@@ -3957,8 +3957,12 @@ static void *cfq_init_queue(struct request_queue *q)
 		return NULL;
 
 	cfqd = kmalloc_node(sizeof(*cfqd), GFP_KERNEL | __GFP_ZERO, q->node);
-	if (!cfqd)
+	if (!cfqd) {
+		spin_lock(&cic_index_lock);
+		ida_remove(&cic_index_ida, i);
+		spin_unlock(&cic_index_lock);
 		return NULL;
+	}
 
 	cfqd->cic_index = i;
 
-- 
1.7.1

